### PR TITLE
gh-101819: Remove _PyWindowsConsoleIO_Type from the Windows DLL

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-101819

Automerge-Triggered-By: GH:erlend-aasland

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Does this work for you, @zooba? Re. https://github.com/python/cpython/issues/101819#issuecomment-1428187675

**Comment by zooba** [comment]
Can you remove the `Check` macro entirely? It's got to go if we're supporting multiple instances of the module, so may as well just go now.

I'd kinda like to see a few more `tstate` parameters being passed around, but I guess if the lower level APIs don't take them then there's nowhere to pass them 🤷‍♂️

**Comment by erlend-aasland** [comment]
> Can you remove the `Check` macro entirely? It's got to go if we're supporting multiple instances of the module, so may as well just go now.

The one in `PC/_testconsole.c`, `Python/pylifecycle.c`, or both? I assume the assert in `Modules/_io/winconsoleio.c` should stay.

**Comment by zooba** [comment]
I was thinking the definition in `_iomodule.h` and anywhere it's used, but I see now that's fully internal anyway.

Provided we're not changing any supported public API, it's fine.

**Comment by erlend-aasland** [comment]
> Provided we're not changing any supported public API, it's fine.

AFAICS, `_iomodule.h` is not included in `Python.h`, so we should be fine.

**Comment by ericsnowcurrently** [comment]
Don't worry about waiting for that PR to merge.  I don't think it conflicts.  Regardless, I don't mind fixing my branch if needed.

File: Modules/_io/_iomodule.h [hunk 1]
```diff
@@ -21,13 +21,10 @@ extern PyTypeObject PyBufferedRandom_Type;
 extern PyTypeObject PyTextIOWrapper_Type;
 extern PyTypeObject PyIncrementalNewlineDecoder_Type;
 
-#ifndef Py_LIMITED_API
 #ifdef MS_WINDOWS
 extern PyTypeObject PyWindowsConsoleIO_Type;
-PyAPI_DATA(PyObject *) _PyWindowsConsoleIO_Type;
-#define PyWindowsConsoleIO_Check(op) (PyObject_TypeCheck((op), (PyTypeObject*)_PyWindowsConsoleIO_Type))
+#define PyWindowsConsoleIO_Check(op) (PyObject_TypeCheck((op), (PyTypeObject*)&PyWindowsConsoleIO_Type))
```
Comment (kumaraditya303) [inline Modules/_io/_iomodule.h]: You can remove this now.

Comment (erlend-aasland) [reply Modules/_io/_iomodule.h]: PTAL

File: Modules/_io/winconsoleio.c [hunk 2]
```diff
@@ -260,7 +260,7 @@ _io__WindowsConsoleIO___init___impl(winconsoleio *self, PyObject *nameobj,
     int fd_is_own = 0;
     HANDLE handle = NULL;
 
-    assert(PyWindowsConsoleIO_Check(self));
+    assert(PyObject_TypeCheck((self), (PyTypeObject *)&PyWindowsConsoleIO_Type));
```
Comment (kumaraditya303) [inline Modules/_io/winconsoleio.c]:
```suggestion
    assert(PyObject_TypeCheck(self, (PyTypeObject *)&PyWindowsConsoleIO_Type));
```

**Review by kumaraditya303 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the reviews; highly appreciated.

**Pull request merged**